### ISSUE TITLE
Why there is both `path` and `path_prefix` for `/files` endpoint?

### ISSUE THREAD

**Issue by yarikoptic** [dandi/dandi-api#422]
![image](https://user-images.githubusercontent.com/39889/126220635-288f87c2-c213-422c-be93-f8006d8e120f.png)
seems to be doing the same thing when I tried.  Should then be only a single parameter, not two .

**Comment by dchiquito** [comment]
Both do the same thing in the code. `path` is provided by an `AssetFilter`, which is the more correct way to approach the problem, while `path_prefix` is used in the code to explicitly query for assets whose path starts with `path_prefix`.

Currently only `path_prefix` works, but I think we should get `path` working correctly and delete `path_prefix`.

**Comment by yarikoptic** [comment]
note that they aren't exactly "the same thing", since `path_prefix` manages to strip itself off whenever it is not ending with trailing `/`:
```
$> curl -s -X GET "https://api.dandiarchive.org/api/dandisets/000027/versions/draft/assets/paths/?path=sub" -H "accept: application/json" | jq .
{
  "folders": {
    "sub-RAT123": {
      "size": 18792,
      "num_files": 1,
      "created": "2021-06-29T22:59:29.678450Z",
      "modified": "2021-06-29T22:59:29.746868Z"
    }
  },
  "files": {}
}

$> curl -s -X GET "https://api.dandiarchive.org/api/dandisets/000027/versions/draft/assets/paths/?path_prefix=sub" -H "accept: application/json" | jq .
{
  "folders": {
    "-RAT123": {
      "size": 18792,
      "num_files": 1,
      "created": "2021-06-29T22:59:29.678450Z",
      "modified": "2021-06-29T22:59:29.746868Z"
    }
  },
  "files": {}
}
```

**Comment by waxlamp** [comment]
I think this has been answered; plans for a fix can continue in #444.
